tools/wpt/metadata.py

```diff
@@ -0,0 +1,155 @@
+import argparse
+import logging
+import os
+import re
+from typing import Any, Optional, Mapping, Sequence
+
+import pydantic
+import yaml
+from pydantic import BaseModel
+
+
+from ..manifest import manifest
+from .virtualenv import Virtualenv
+
+here = os.path.dirname(__file__)
+wpt_root = os.path.abspath(os.path.join(here, os.pardir, os.pardir))
+
+class UrlResultsModel(BaseModel):
+    model_config = pydantic.ConfigDict(extra='forbid')
```

Comment:
@jcscottiii FYI, I didn't know about Pydantic before, but it's possible we could use it for validating WEB_FEATURES.yml files. Don't fix what ain't broke I guess, but maybe if there's a need to change it in the future.